internal/controller/memcached_controller.go

```diff
@@ -202,3 +312,68 @@ func getPodNames(pods []corev1.Pod) []string {
 
 	return names
 }
+
+// isValidVersion queries the docker hub API to check if the provided version can be resolved to an image tag.
+func isValidVersion(version string) (bool, error) {
+	// Check if version is valid Semver
+	_, err := goversion.NewSemver(version)
+	if err != nil {
+		// Assume error means the version is malformed.
+		return false, nil
+	}
+
+	// Check if tag exists in docker hub
+	tag := imageTag(version)
+	url := fmt.Sprintf("http://hub.docker.com/v2/repositories/library/memcached/tags/%s", tag)
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Println("Error checking tag:", err)
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	fmt.Println("Got status:", resp.StatusCode)
+
+	if resp.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return true, nil
+	}
+
```

Comment:
This is a nice touch! I think you're the first one to do this task that has thought to check for availability